scripts/memrem.coffee

```diff
@@ -10,30 +10,55 @@
 # Commands:
 #   hubot mem <keyword> - prints out saved resource
 #   hubot get <keyword> - prints out saved resource
+#   hubot get list - prints out saved keys
 #   hubot rem <keyword> <resource> - saves resource to specified keyword
 #   hubot set <keyword> <resource> - saves resource to specified keyword
 #
 # Author:
 #   snibbles
 
-memremId = (keyword) -> "memrem_#{keyword}"
+memremBrain = "memrem"
 
 onGet = (msg) ->
-  keyword = msg.match[2]
-  key = memremId keyword
-  value = msg.robot.brain.get key
-  if value
+  key = msg.match[2]
+
+  if key is "list"
+    return
+
+  if value = brainMatch(msg.robot.brain, key)
     msg.send value
   else
-    msg.send "we does not has #{keyword}"
+    msg.send "Couldn't find #{key}."
+
+brainMatch = (brain, key) ->
+  storage = brain.get memremBrain
+
+  if !storage
+    brain.set memremBrain, {}
+
+  return storage[key]
 
 onSet = (msg) ->
-  keyword = msg.match[2]
+  key = msg.match[2]
+  if key is "list"
+    return
+
   value = msg.match[3]
-  key = memremId keyword
-  msg.robot.brain.set key, value
-  msg.send "saved #{value} as #{keyword}"
+  brain = msg.robot.brain
+  storage = brain.get memremBrain
```

Comment:
I think

```coffee
storage = brain.get memremBrain

if !storage
  storage = {}
```

could be transformed into

```coffee
storage = brain.get memremBrain ? {}
```

via http://coffeescript.org/#operators (specifically the existential operator)